firebase-test/src/documents-rules.test.ts

```diff
@@ -159,6 +159,140 @@ describe("Firestore security rules", () => {
 
   });
 
+  describe("history entries", () => {
+    const kDocumentDocPath = "authed/myPortal/documents/myDocument";
+    const kDocumentHistoryDocPath = `${kDocumentDocPath}/history/myHistoryEntry`;
+    interface ISpecHisoryDoc {
+      add?: Record<string, string | string[] | object>;
+      remove?: string[];
+    }
+    function specHistoryEntryDoc(options?: ISpecHisoryDoc) {
+      // a valid history document specification
+      const historyDoc = { id: "an-id", tree: "my-document", action: "/content/stuff",
+                           undoable: true, createdAt: mockTimestamp(),
+                           records: [], state: "complete"
+                         };
```

Comment:
I'm pretty sure the firestore document for a history entry looks like:
```
{
  index: [number], // used to be a firestore timestamp
  entry: JSON.stringify(historyDoc),
  created: [firestore timestamp],
  previousEntryId: [string or null]
}
```

I'd guess this doesn't actually mater because the rules are not checking this firestore document. But is confusing to see this wrong.